### PR TITLE
Branch and Terminator enums

### PR DESCRIPTION
**Pull request by kvark**
This is the second half of #86 with the review notes addressed.
"//TODO" comments don't show up in the autogenerated code, I only put them into the autogen itself. We'll need to figure out what to do with `Context` before turning them into real `Token<BasicBlock>` and such (since today context doesn't know about basic blocks).

### CODE REVIEW

File: autogen/sr.rs [hunk 1]
```diff
@@ -38,97 +38,122 @@ impl OperandTokens {
...
+                    ),
+                    "Parameter Types" => (
+                        quote! { Token<Type> },
+                        quote! { self.types.lookup(*value).unwrap() },
+                    ),
+                    name if name.ends_with(" Type") => (
+                        quote! { Token<Type> },
+                        quote! { self.types.lookup(*value).unwrap() },
+                    ),
+                    //TODO:
```
Comment (antiagainst) [inline autogen/sr.rs]: Super nit: Typically we leave a space between `//` and the following text in comments in C++ world. Is this not true for Rust?

Comment (kvark) [reply autogen/sr.rs]: Typically in Rust we are leaving a space after `//`. However, for special notes that start with TODO, FIXME, and so forth, I'm not putting the space in, which in part indicates that these comments are special. I'm fine with changing this, just explaining the general approach I've been doing in other projects.

**Comment by kvark** [comment]
> But we should consider coupling changes with tests as early as possible once we have enough infra.

As soon as we know that we are on the right path, I'll be happy to add more tests.

**Comment by kvark** [comment]
Going to wait for #93 and #94 to be reviewed first, in case they need to be merged - I don't ant to cause @AIOOB any rebase headaches :)

File: autogen/sr.rs (hunk 1 above)
Comment (antiagainst) [reply autogen/sr.rs]: Sure. I don't care that much on this.

**Comment by kvark** [comment]
This was reviewed and approved. Rebasing didn't involve any structural changes. Merging.

**Pull request merged**